addons/ballistics/functions/fnc_statTextStatement_weaponMuzzleVelocity.sqf

```diff
@@ -4,52 +4,49 @@
  * Text statement for the weapon muzzle velocity stat.
  *
  * Arguments:
- * 0: Type what it is here <TYPE> (unused)
- * 1: Item config path <CONFIG>
+ * 0: Stats <ARRAY> (unused)
+ * 1: Weapon config path <CONFIG>
+ * 2: Magazine config path <CONFIG> (default: configNull)
  *
  * Return Value:
  * Display text <STRING>
  *
  * Public: No
 */
 
-params ["", "_configWeapon"];
+params ["", "_configWeapon", ["_configMagazine", configNull]];
 
-if (EGVAR(arsenal,currentLeftPanel) == 2002) then {
-    private _primaryMag = primaryWeaponMagazine EGVAR(arsenal,center);
-    [primaryWeapon EGVAR(arsenal,center), _primaryMag param [0, ""]]
-} else {
-    private _primaryMag = handgunMagazine EGVAR(arsenal,center);
-    [handgunWeapon EGVAR(arsenal,center), _primaryMag param [0, ""]]
-} params ["_weapon", "_magazine"];
+if (isNull _configMagazine) then { // Get the current weapon's mag
+    _configMagazine = configFile >> "CfgMagazines" >> (weaponState EGVAR(arsenal,center)) select 3;;
+};
 
-if (_magazine isEqualTo "") then {
+if (isNull _configMagazine) exitWith { // Nothing in the magwell
     localize "str_empty";
+};
+
+private _muzzleVelocity = getNumber (_configMagazine >> "initSpeed");
+
+private _abAdjustText = "";
+private _useAB = missionNamespace getVariable [QEGVAR(advanced_ballistics,enabled), false] && {EGVAR(advanced_ballistics,barrelLengthInfluenceEnabled)};
```

Comment:
Boolean checks are too fast for lazy eval. If anything, the order should be flipped:
```suggestion
private _useAB = EGVAR(advanced_ballistics,barrelLengthInfluenceEnabled) && {missionNamespace getVariable [QEGVAR(advanced_ballistics,enabled), false]};
```